src/esp/sensor/FisheyeSensor.cpp

```diff
@@ -155,12 +155,36 @@ bool FisheyeSensor::drawObservation(sim::Simulator& sim) {
 
   // generate the cubemap texture
   if (fisheyeSensorSpec_->sensorType == SensorType::Semantic) {
-    cubeMap_->renderToTexture(*cubeMapCamera_,
-                              sim.getActiveSemanticSceneGraph(), flags);
-    if (&sim.getActiveSemanticSceneGraph() != &sim.getActiveSceneGraph()) {
+    bool twoSceneGraphs =
+        (&sim.getActiveSemanticSceneGraph() != &sim.getActiveSceneGraph());
+
+    if (twoSceneGraphs) {
+      // *assume* the sensor is always in the regular scene graph.
+      // backup first
+      Mn::Matrix4 relativeTransform = node().transformation();
```

Comment:
Perhaps this should be abstracted out if possible. Disappointing that this hack is necessary, but it is required for every VisualSensor, right?

Reply:
That can also be saved for a later PR though.

Reply:
Yes, a bit code duplication here. I did not figure out a good way to extract it so far. Like I said in the description, different visual sensors has it is own rendering logic. Read this two pieces of code. They are a bit different.

Reply:
Allow me to think about it.
This is problem that we need to solve.

Otherwise it will become a big problem as more sensors are coming in the future.

@mosra @aclegg3 @erikwijmans any ideas here?

Reply:
I am thinking the ways to address this. It must be solved before committing this diff in.